### ISSUE TITLE
compile error at heimdal-gssapi>=1.6.0

### ISSUE THREAD

**Issue by zingaro1972** [greenbone/openvas-smb#42]
Hi, 
the compile with cmake return this error:
-- Could NOT find Git (missing: GIT_EXECUTABLE)
-- Install prefix: /usr/local
-- Checking for module 'heimdal-gssapi>=1.6.0'
--   Package 'heimdal-gssapi', required by 'virtual:world', not found
CMake Error at /usr/share/cmake/Modules/FindPkgConfig.cmake:545 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPkgConfig.cmake:733 (_pkg_check_modules_internal)
  CMakeLists.txt:147 (pkg_check_modules)

I have the files into the  /usr/lib64/heimdal ...

this is the debug installation: Running with debug output on for the `find` commands.
CMake Debug Log at /usr/share/cmake/Modules/FindGit.cmake:54 (find_program):
  find_program called with the following settings:

    VAR: GIT_EXECUTABLE
    NAMES: "git"
           "eg"
    Documentation: Git command line client
    Framework
      Only Search Frameworks: 0
      Search Frameworks Last: 0
      Search Frameworks First: 0
    AppBundle
      Only Search AppBundle: 0
      Search AppBundle Last: 0
      Search AppBundle First: 0
    CMAKE_FIND_USE_CMAKE_PATH: 1
    CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH: 1
    CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH: 1
    CMAKE_FIND_USE_CMAKE_SYSTEM_PATH: 1

  find_program considered the following locations:

    /usr/local/bin/git
    /usr/bin/git
    /usr/local/sbin/git
    /usr/sbin/git
    /usr/lib64/git
    /usr/local/git
    /usr/git
    /bin/git
    /sbin/git
    /usr/X11R6/bin/git
    /usr/X11R6/sbin/git
    /usr/X11R6/git
    /usr/pkg/bin/git
    /usr/pkg/sbin/git
    /usr/pkg/git
    /opt/bin/git
    /opt/sbin/git
    /opt/git
    /usr/local/bin/eg
    /usr/bin/eg
    /usr/local/sbin/eg
    /usr/sbin/eg
    /usr/lib64/eg
    /usr/local/eg
    /usr/eg
    /bin/eg
    /sbin/eg
    /usr/X11R6/bin/eg
    /usr/X11R6/sbin/eg
    /usr/X11R6/eg
    /usr/pkg/bin/eg
    /usr/pkg/sbin/eg
    /usr/pkg/eg
    /opt/bin/eg
    /opt/sbin/eg
    /opt/eg

how i can add the correct path to cmake ?

**Comment by Dexus** [comment]
Hi @zingaro1972 you need to change some more things for alpine linux. Please know that we currently only support debian buster for this version.

**Comment by tuxvador** [comment]
Managed to install heimdal-gssapi on arch linux for openvas 21.0.4.
Will try to make a package for openvas 21.4.4

**Comment by guillaumechardin** [comment]
I tried to buidl this on almalinux 9.2 too. It fails with the same error.
I successfully build the heimdal package, so 
1. what are the file searched for heimdal to successfully paass this test ?
2. how to specify a new path to get cmake find the package and make it succeed compiling ?
   for that i tried these two settings in CMakeLists.txt : 
     include_directories("/usr/heimdal/include") 
     add_subdirectory (/usr/heimdal/include/gssapi)

thanks

**Comment by guillaumechardin** [comment]
For those facing this issue ie in almalinux, this is somewhat pkg-config related.
Some heimdal files are wrongly placed.

run  first :
pkg-config --list-all #heimdal-gssapi should not be listed

2. check heimdal-gssapi.pc file location
find / -type f | grep heimdal-gssapi.pc   #in my case /usr/lib64/heimdal/lib/pkgconfig/

# cd  /usr/lib64/pkgconfig/   #right location for pc files in my distro, check too
#  ln -s /usr/lib64/heimdal/lib/pkgconfig/heimdal-gssapi.pc   #create symlink in right location :)

pkg-config --list-all  #now should be listed and cmake will pass

**Comment by amin-prog** [comment]
I have the same issue on
Kali GNU/Linux Rolling x86_64  
Kernel: 6.4.0-kali3-amd64

`$ dpkg-query -s heimdal-gssapi                                         
dpkg-query: package 'heimdal-gssapi' is not installed and no information is available
Use dpkg --info (= dpkg-deb --info) to examine archive files.
`

Please, any solution?

**Comment by guillaumechardin** [comment]
> I have the same issue on Kali GNU/Linux Rolling x86_64 Kernel: 6.4.0-kali3-amd64
> 
> `$ dpkg-query -s heimdal-gssapi
> 
> dpkg-query: package 'heimdal-gssapi' is not installed and no information is available Use dpkg --info (= dpkg-deb --info) to examine archive files.`
> 
> Please, any solution?

Did you apply my solution (see previous post ?)

**Comment by amin-prog** [comment]
> > I have the same issue on Kali GNU/Linux Rolling x86_64 Kernel: 6.4.0-kali3-amd64
> > `$ dpkg-query -s heimdal-gssapi
> > dpkg-query: package 'heimdal-gssapi' is not installed and no information is available Use dpkg --info (= dpkg-deb --info) to examine archive files.`
> > Please, any solution?
> 
> Did you apply my solution (see previous post ?)

Yes, but 'find' (step 2) did'nt find anything... This was the output that I got :
`$ sudo find / -type f | grep hemidal-gssapi.pc
find: '/run/user/1000/doc': Permission denied
`
I don't know whats wrong.

**Comment by amin-prog** [comment]
> > I have the same issue on Kali GNU/Linux Rolling x86_64 Kernel: 6.4.0-kali3-amd64
> > `$ dpkg-query -s heimdal-gssapi
> > dpkg-query: package 'heimdal-gssapi' is not installed and no information is available Use dpkg --info (= dpkg-deb --info) to examine archive files.`
> > Please, any solution?
> 
> Did you apply my solution (see previous post ?)

Yes, but 'find' (step 2) did'nt find anything... This was the output that I got :
`$ sudo find / -type f | grep hemidal-gssapi.pc \n
find: '/run/user/1000/doc': Permission denied
`
I don't know whats wrong.

**Comment by amin-prog** [comment]
> > Did you apply my solution (see previous post ?)
> 
> Yes, but 'find' (step 2) did'nt find anything... This was the output that I got : `$ sudo find / -type f | grep hemidal-gssapi.pc
> 
> find: '/run/user/1000/doc': Permission denied ` I don't know whats wrong.

Now I see, the package isn't even installed. So I will have to install it first.

**Comment by olyop** [comment]
For Fedora I followed the instructions at [https://unix.stackexchange.com/a/1344](https://unix.stackexchange.com/a/1344) and this error went away and for Ubuntu I assume it would be `sudo apt-get install build-essential`.

**Comment by r3k2** [comment]
I am having this issue as well on Arch GNU/Linux mostly related because openvas-smb looks for older heimdal-gssapi instead of krb5 package, I decided to get my hands at work and I compiled (with a lot of effort and research) the older heimdal  the make install as mention above will install to /usr/heimdal/   then I have to link all  the pkgconfig to the right place for Arch GNU/Linux but even like this it will not finish compiling :(   I will post latest the output is in my lab and I am out now for a coffee.
